core/src/main/scala/org/apache/spark/metrics/source/MemorySource.scala

```diff
@@ -0,0 +1,145 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.metrics.source
+
+import com.codahale.metrics.MetricRegistry
+
+import org.apache.spark.executor.ExecutorMetrics
+
+private[spark] class MemorySource extends Source {
+
+  override val metricRegistry = new MetricRegistry()
+  override val sourceName = "MemoryMetrics"
+
+  val METRIC_JVM_HEAP_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("JVMHeapMemory"))
+  val METRIC_JVM_OFF_HEAP_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("JVMOffHeapMemory"))
+  val METRIC_JVM_ON_HEAP_EXECUTION_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OnHeapExecutionMemory"))
+  val METRIC_JVM_ON_HEAP_STORAGE_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OnHeapStorageMemory"))
+  val METRIC_JVM_ON_HEAP_UNIFIED_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OnHeapUnifiedMemory"))
+  val METRIC_JVM_OFF_HEAP_EXECUTION_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OffHeapExecutionMemory"))
+  val METRIC_JVM_OFF_HEAP_STORAGE_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OffHeapStorageMemory"))
+  val METRIC_JVM_OFF_HEAP_UNIFIED_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("OffHeapUnifiedMemory"))
+  val METRIC_JVM_DIRECT_POOL_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("DirectPoolMemory"))
+  val METRIC_JVM_MAPPED_POOL_MEMORY =
+    metricRegistry.counter(MetricRegistry.name("MappedPoolMemory"))
+
+  // GarbageCollectionMetrics
+  val METRIC_MINOR_GC_COUNT =
+    metricRegistry.counter(MetricRegistry.name("MinorGCCount"))
+  val METRIC_MINOR_GC_TIME =
+    metricRegistry.counter(MetricRegistry.name("MinorGCTime"))
+  val METRIC_MAJOR_GC_COUNT =
+    metricRegistry.counter(MetricRegistry.name("MajorGCCount"))
+  val METRIC_MAJOR_GC_TIME =
+    metricRegistry.counter(MetricRegistry.name("MajorGCTime"))
+
+  // ProcessTreeMetrics
+  val METRIC_PROCESS_TREE_JVM_VMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreeJVMVMemory"))
+  val METRIC_PROCESS_TREE_JVM_RSSMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreeJVMRSSMemory"))
+  val METRIC_PROCESS_TREE_PYTHON_VMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreePythonVMemory"))
+  val METRIC_PROCESS_TREE_PYTHON_RSSMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreePythonRSSMemory"))
+  val METRIC_PROCESS_TREE_OTHER_VMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreeOtherVMemory"))
+  val METRIC_PROCESS_TREE_OTHER_RSSMEMORY =
+    metricRegistry.counter(MetricRegistry.name("ProcessTreeOtherRSSMemory"))
+
+  def updateCounters(memorySource: MemorySource,
```

Comment:
The updateCounters is a member function and it should use the counters of the its own instance so the `memorySource` argument is not needed.

This way `memorySource.updateCounters(memorySource, executorUpdates)` will be just `memorySource.updateCounters(executorUpdates)`

Reply:
Indeed. Thanks for the comment.